pennylane/devices/default_tensor.py

```diff
@@ -0,0 +1,599 @@
+        "Prod",
+        "Exp",
+    }
+)
+# The set of supported observables.
+
+_methods = frozenset({"mps"})
+# The set of supported methods.
+
+
+def accepted_methods(method: str) -> bool:
+    """A function that determines whether or not a method is supported by ``default.tensor``."""
+    return method in _methods
+
+
+def stopping_condition(op: qml.operation.Operator) -> bool:
+    """A function that determines if an operation is supported by ``lightning.tensor`` for this interface."""
+    return op.name in _operations
+
+
+def accepted_observables(obs: qml.operation.Operator) -> bool:
+    """A function that determines if an observable is supported by ``lightning.tensor`` for this interface."""
+    return obs.name in _observables
+
+
+@simulator_tracking
+@single_tape_support
+class DefaultTensor(Device):
+    """A PennyLane device to perform tensor network operations on a quantum circuit using
+    `quimb <https://github.com/jcmgray/quimb/>`_.
+
+    Args:
+        wires (int, Iterable[Number, str]): Number of wires present on the device, or iterable that
+            contains unique labels for the wires as numbers (i.e., ``[-1, 0, 2]``) or strings
+            (``['aux_wire', 'q1', 'q2']``). Default is ``None``.
+        shots (int, Sequence[int], Sequence[Union[int, Sequence[int]]]): The default number of shots
+            to use in executions involving this device. Currently, it can only be ``None``, so that computation of
+            statistics like expectation values and variances is performed analytically.
+        method (str): Supported method. Currently, only ``mps`` is supported.
+        dtype (type): Datatype for the tensor representation. Must be one of ``np.complex64`` or ``np.complex128``.
+            Default is ``np.complex128``.
+        **kwargs: keyword arguments. The following options are currently supported:
+
+            ``max_bond_dim`` (int): Maximum bond dimension for the MPS simulator.
+                It corresponds to the number of Schmidt coefficients retained at the end of the SVD algorithm when applying gates. Default is ``None``.
+            ``cutoff`` (float): Truncation threshold for the Schmidt coefficients in a MPS simulator. Default is ``np.finfo(dtype).eps``.
+            ``contract`` (str): The contraction method for applying gates. It can be either ``auto-mps`` or ``nonlocal``.
+                ``nonlocal`` turns each gate into a MPO and applies it directly to the MPS, while ``auto-mps`` swaps nonlocal qubits in 2-qubit gates to be next
+                    to each other before applying the gate, then swaps them back. Default is ``auto-mps``.
+    """
+
+    # pylint: disable=too-many-instance-attributes
+
+    # So far we just consider the options for MPS simulator
+    _device_options = (
+        "contract",
+        "cutoff",
+        "dtype",
+        "method",
+        "max_bond_dim",
+    )
+
+    _new_API = True
+
+    def __init__(
+        self,
+        wires=None,
```

Comment:
```suggestion
        wires,
```

Looks like we need to make wires mandatory for this device.  If we provide `wires=None` , we end up getting:
```
File [~/Prog/pennylane/pennylane/devices/default_tensor.py:354](http://localhost:8889/~/Prog/pennylane/pennylane/devices/default_tensor.py#line=353), in DefaultTensor.execute(self, circuits, execution_config)
    350 results = []
    351 for circuit in circuits:
    352     # we need to check if the wires of the circuit are compatible with the wires of the device
    353     # since the initial tensor state is created with the wires of the device
--> 354     if not self.wires.contains_wires(circuit.wires):
    355         raise AttributeError(
    356             f"Circuit has wires {circuit.wires.tolist()}. "
    357             f"Tensor on device has wires {self.wires.tolist()}"
    358         )
    359     circuit = circuit.map_to_standard_wires()

AttributeError: 'NoneType' object has no attribute 'contains_wires'
```

Reply:
Thanks for noticing this @albi3ro.
For context, it was part of the requirements of the first epic not to raise an error if the `wire` argument is not provided. On the other hand, if we have a mismatch between wires on the circuit and the device, `quimb` raises a very cryptic error, so I had to intercept this with the above `if` condition (causing the issue you caught). 
Furthermore, when we create the initial state with `quimb` we *need* to provide a set of wires, so I agree that in the end the best solution is to make `wires` a positional mandatory argument.